### PR TITLE
CloudWatch Logs - Subscription Filter

### PR DESCRIPTION
**Pull request by bdwyertech**
Add a filter for CWL SubscriptionFilters

Within my organization, we often set up forwarding to Splunk.  This filter would allow us to look for and modify CWL groups with/without subscriptions.

### CODE REVIEW

File: c7n/resources/cw.py [hunk 1]
```diff
@@ -705,6 +705,59 @@ def process_resource_set(self, client, accounts, resources):
...
+                    value: arn:aws:lambda:us-east-1:123456789876:function:forwarder
+    """
+    schema = type_schema('subscription-filter', rinherit=ValueFilter.schema)
+    annotation_key = 'c7n:SubscriptionFilters'
+    permissions = ('logs:DescribeSubscriptionFilters',)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('logs')
+        results = []
+        with self.executor_factory(max_workers=1) as w:
```
Comment (kapilt) [inline c7n/resources/cw.py:731]: given max workers 1 we can just walk through the set without the executor factory.

Comment (bdwyertech) [reply c7n/resources/cw.py:731]: OK. I stole that from from cross-account filter. Would you like me to fix that one too?

Comment (bdwyertech) [reply c7n/resources/cw.py]: Resolved by 8d45c5ba403c8e72ee8c98cacc63d250e5578bb6

File: c7n/resources/cw.py [hunk 2]
```diff
@@ -728,19 +728,14 @@ class LogSubscriptionFilter(ValueFilter):
...
-                    w.submit(
-                        self.process_resource_set, client, rset))
-            for f in as_completed(futures):
-                if f.exception():
-                    self.log.error(
-                        "Error checking log groups subscription-filters %s",
-                        f.exception())
-                    continue
-                results.extend(f.result())
+        for rset in chunks(resources, 50):
```
Comment (bdwyertech) [inline c7n/resources/cw.py:731]: Maybe we don't need this chunking logic either -- wyt? @kapilt

Comment (bdwyertech) [reply c7n/resources/cw.py]: Removed it in ad6e310

**Review by kapilt — APPROVED** [review]
lgtm, thanks

**Pull request merged**